src/check.c

```diff
@@ -244,18 +244,31 @@ static time_t date_dos2unix(unsigned short time, unsigned short date)
 {
     int month, year;
     time_t secs;
+    long long int tempsecs;
 
     month = ((date >> 5) & 15) - 1;
     if (month < 0) {
 	/* make sure that nothing bad happens if the month bits were zero */
 	month = 0;
     }
     year = date >> 9;
-    secs =
+    tempsecs =
 	(time & 31) * 2 + 60 * ((time >> 5) & 63) + (time >> 11) * 3600 +
-	86400 * ((date & 31) - 1 + day_n[month] + (year / 4) + year * 365 -
+	(long long int)86400 * ((date & 31) - 1 + day_n[month] + (year / 4) + year * 365 -
 		 ((year & 3) == 0 && month < 2 ? 1 : 0) + 3653);
     /* days since 1.1.70 plus 80's leap day */
+
+    /* "time_t" is specified as "any arithmetic type" by POSIX, it can be as
+       small as a signed integer */
+    if (tempsecs > INT32_MAX) {
```

Comment:
This is not a correct check. Type `time_t` can be also floating-point...

Also on more systems `time_t` is 64bit therefore limiting it to just 32bit is not a good idea.

Reply:
Theoretically, you're right. In practice, no system ever used a floating-point type, and POSIX.1-2013 now requires time_t to be an integer type.

Also, time_t is 32-bit on some platforms (hence the upcoming Year 2038 bug). I agree that the check could be refined based on sizeof(time_t), but one can also hope that MS-DOS will have completely disappeared by year 2038 :)